WcaOnRails/app/controllers/api/v0/api_controller.rb

```diff
@@ -9,9 +10,17 @@ def bad_api_parameter(e)
 
   DEFAULT_API_RESULT_LIMIT = 20
 
+  def current_resource_owner
+    if doorkeeper_token
+      User.find_by_id(doorkeeper_token.resource_owner_id).tap do |user|
+        user.doorkeeper_token = doorkeeper_token
```

Comment:
3. Not really, in case the user is not found `#find_by_id` or `find_by(id: ...)` returns `nil` (good in this case), whereas `#find` throws an exception.